lib/candy_check/app_store/receipt_collection.rb

```diff
@@ -10,13 +10,15 @@ class ReceiptCollection
       # from Apple's verification server
       # @param attributes [Array<Hash>]
       def initialize(hash: nil, mapping: nil)
-        if (!mapping.nil?)
-          @receipts = mapping.map do |r|
+        receipts = if (!mapping.nil?)
+          mapping.map do |r|
             Receipt.new(r)
           end
         else
-          @receipts = [Receipt.new(hash)]
-        end
+          [Receipt.new(hash)]
+       end
+        # Sort by purchase date ascending: Older to most recent. Apple _should_ already be sending that
+        @receipts = receipts.sort_by!(&purchase_date).reverse!
```

Comment:
The `!` performs an in-place operation, which does not apper to be needed in this context

```suggestion
        @receipts = receipts.sort_by(&purchase_date).reverse
```